### ISSUE TITLE
[Chapter 03] SGDClassifier Making Incorrect Prediction

### ISSUE THREAD

**Issue by mauricio-ms** [ageron/handson-ml2#123]
Hi Aurélien,

The following code is predicting the number 3 instead 5:

`sgd_clf.fit(X_train, y_train)
sgd_clf.predict([some_digit])`

I compare with the notebook in this repository and used the same parameters to create the model, but the error persists.

I have found other differences as in the _precision and recall versus the decision threshold plot_ and some scores produced along the chapter.

Maybe it's due to a difference in versions of some library, to start the code I used the `requirements.txt` to install the dependencies.

**Note:** I see that in your notebook at cell 15, the line `sgd_clf.fit(X_train, y_train_5)` returns the model with parameter `n_iter=None`, and in my notebook this parameter is not displayed.

**Comment by mauricio-ms** [comment]
Installing the sklearn version 0.20.0 resolve all the problems, I get the same results as the book in the entire notebook.

**Comment by aaronwhitnee** [comment]
Thanks for sharing your solution to this problem @mauricio-ms!

I ran into this same issue when using sklearn version 0.22.2. In my `requirements.txt`, I replaced `scikit-learn==0.22` with `scikit-learn==0.20.0`, re-installed scikit-learn (at version 0.20.0), restarted the Jupyter kernel (and waited for all the models to re-train... *sigh*), and then I got the same prediction as the book!

As a sanity check, before rerunning my whole notebook, I ran this to confirm the correct version was installed (from within the notebook):
`>>> import sklearn`
`>>> sklearn.__version__ # outputs the installed version of the library`
